src/Kanbas/Courses/Quizzes/List/index.tsx

```diff
@@ -1,22 +1,157 @@
-import QuizDetails from "../Details";
+import { useDispatch, useSelector } from "react-redux";
 import { useNavigate, useParams } from "react-router";
+import { KanbasState } from "../../../store";
+import { useEffect, useState } from "react";
+import { IKanbasQuiz } from "../../../store/interfaces/quizzes";
+import { setQuizzes, toggleQuizPublished } from "../reducer";
+import {
+  FaBan,
+  FaCaretDown,
+  FaCheckCircle,
+  FaEllipsisV,
+  FaRocket,
+} from "react-icons/fa";
+import { Collapse } from "react-bootstrap";
+import { getCourseQuizzes, postQuizSetPublish } from "../client";
+import "./index.css";
 
 function QuizList() {
+  const { courseId } = useParams();
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
 
-    const courseId = useParams();
-    const navigate = useNavigate();
+  const quizzesList = useSelector(
+    (state: KanbasState) => state.quizzesReducer.quizzes
+  );
 
-    const handleQuizSelection = () => {
-        // TODO pass quiz id later to this function and update the path to direct to QuizDetails
-        navigate(`/Kanbas/Courses/${courseId}/Quizzes/`);
+  const [quizzesExpanded, setQuizzesExpanded] = useState(true);
+
+  const handleQuizSelection = (quizId: string) => {
+    navigate(`/Kanbas/Courses/${courseId}/Quizzes/${quizId}`);
+  };
+
+  const handlePublishToggle = (quizId: string) => {
+    const _isPublished =
+      quizzesList.find((q) => q._id == quizId)?.isPublished ?? false;
+
+    postQuizSetPublish(quizId, !_isPublished).then(() => {
+      dispatch(toggleQuizPublished(quizId));
+    });
+  };
+
+  const getDateDisplay = (date: Date) => {
```

Comment:
Do you still need this?

Reply:
I am keeping this for later, but yes, we need it. I'll move it to utils once I'm done